### PR TITLE
Fixes thirst not applying speed debuffs properly , Makes Thirst a server-side component only

### PR DESCRIPTION
**Pull request by MLGTASTICa**
## About the PR 
All logic functionality removed from ThirstComponent and moved to ThirstSystem
Thirst damage no longer scales with frametime, since frames are calculated by the second
Removed client-side and shared ThirstComponents , since they provided no functionality(and there was no reason to send network updates for thirst anyway)

**Screenshots**

**Changelog**

:cl: SPCR
- fix : Fixed being extremly thirsty not applying its slowdown properly

### CODE REVIEW

File: Content.Server/Nutrition/EntitySystems/ThirstSystem.cs [hunk 1]
```diff
@@ -1,14 +1,126 @@
...
+                (int) component.ThirstThresholds[ThirstThreshold.Okay] - 1);
+            component.CurrentThirstThreshold = GetThirstThreshold(component, component.CurrentThirst);
+            component.LastThirstThreshold = ThirstThreshold.Okay; // TODO: Potentially change this -> Used Okay because no effects.
+            // TODO: Check all thresholds make sense and throw if they don't.
+            UpdateEffects(component);
+        }
+
+        private void OnRefreshMovespeed(EntityUid uid, ThirstComponent component, RefreshMovementSpeedModifiersEvent args)
+        {
+            float mod = component.CurrentThirstThreshold == ThirstThreshold.Parched ? 0.75f : 1.0f;
```
Comment (metalgearsloth) [inline Content.Server/Nutrition/EntitySystems/ThirstSystem.cs:46]: Should also check for dead.

**Review by metalgearsloth — CHANGES_REQUESTED** [review]
All of the systems need caching as dependencies, same with logger sawmill.

File: Content.Server/Nutrition/EntitySystems/ThirstSystem.cs [hunk 2]
```diff
@@ -1,14 +1,126 @@
...
+using Content.Shared.Database;
+using Content.Shared.Damage;
 
 namespace Content.Server.Nutrition.EntitySystems
 {
     [UsedImplicitly]
     public sealed class ThirstSystem : EntitySystem
     {
+
+        [Dependency] private readonly IEntityManager _entMan = default!;
```
[the remark this answers is not in the document]
Comment (MLGTASTICa) [reply Content.Server/Nutrition/EntitySystems/ThirstSystem.cs]: Fixed

File: Content.Server/Nutrition/EntitySystems/ThirstSystem.cs (hunk 1 above)
Comment (metalgearsloth) [reply Content.Server/Nutrition/EntitySystems/ThirstSystem.cs:56]: I meant the threshold

File: Content.Server/Nutrition/Components/ThirstComponent.cs [hunk 3]
```diff
@@ -1,33 +1,17 @@
...
-        OverHydrated,
-        Okay,
-        Thirsty,
-        Parched,
-        Dead,
+        Dead = 0,
+        Parched = 1 << 0,
+        Thirsty = 1 << 1,
+        Okay = 1 << 2,
+        OverHydrated = 1 << 3,
```
[the remark this answers is not in the document]
Comment (metalgearsloth) [reply Content.Server/Nutrition/Components/ThirstComponent.cs:14]: Mainly so rider doesn't yell at me when I (A | B)

**Pull request merged**